Выполнил ТЗ 14 спринта

DmitriyStepanov77 opened the pull request.

GalkinAndStepanov committed:
Выполнил ТЗ 14 спринта.

Sla-als commented on src/main/java/ru/practicum/shareit/exeception/model/ConflictExeception.java:
Ошибки в названии: 

ConflictExeception → ConflictException
NotFoundExeception → NotFoundException

Sla-als commented on src/main/java/ru/practicum/shareit/user/service/UserServiceImp.java:
```diff
@@ -0,0 +1,55 @@
+        userStorage.deleteUser(id);
+    }
+
+    private void validationUser(User user) {
```
Валидация в UserServiceImp проверяет наличие email, но эта проверка осуществляется методом isEmpty(), что не учитывает случаи, когда email может быть null. Лучше использовать StringUtils.hasText() или проверку null перед isEmpty().

Sla-als replied:
Или можно использовать аннотации на уровне DTO

Sla-als commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,41 @@
+
+    @PatchMapping("/{id}")
+    public UserDto updateUser(@Valid @RequestBody User user, @PathVariable int id) {
+        return userService.updateUser(id, user);
```
В методе не проверяется, существует ли пользователь с данным id, что может привести к NullPointerException.

Sla-als replied:
В методе ItemServiceImp.updateItem также

Sla-als commented on src/test/java/ru/practicum/shareit/ItemControllerTest.java:
Тесты покрывают основные сценарии, но следовало бы добавить негативные тесты для методов update и delete.

Sla-als reviewed (commented):
Есть замечания:

GalkinAndStepanov committed:
Исправил ошибки в наименованиях exсeption.
Добавил проверку на NULL при проверке email и поиске.
Добавил тесты на апдейт user и item, а также на удаление user.
Исправил присваивание id у item.
Добавил выброс NotFoundException при отсутсвии item.

GalkinAndStepanov committed:
Исправил чекстайл

Sla-als commented:
Отличная работа!
